presto-spi/src/main/java/io/prestosql/spi/type/TimeZoneKey.java

```diff
@@ -210,102 +251,89 @@ public static boolean isUtcZoneId(String zoneId)
-        else if (length > 2 && zoneId.startsWith("ut")) {
-            zoneId = zoneId.substring(2);
-            length = zoneId.length();
         }
 
-        // (+/-)00:00 is UTC
-        if ("+00:00".equals(zoneId) || "-00:00".equals(zoneId)) {
-            return "utc";
+        // Normalize using JDK rules. In particular, normalize forms such as +0735, +7, +073521
+        try {
+            zoneId = ZoneId.of(zoneId).getId();
         }
-
-        // if zoneId matches XXX:XX, it is likely +HH:mm, so just return it
-        // since only offset time zones will contain a `:` character
-        if (length == 6 && zoneId.charAt(3) == ':') {
+        catch (Exception e) {
             return zoneId;
         }
 
-        //
-        // Rewrite (+/-)H[H] to (+/-)HH:00
-        //
-        if (length != 2 && length != 3) {
-            return originalZoneId;
+        if (isUtcEquivalentName(zoneId)) {
+            return "utc";
         }
 
-        // zone must start with a plus or minus sign
-        char signChar = zoneId.charAt(0);
-        if (signChar != '+' && signChar != '-') {
-            return originalZoneId;
-        }
-        if (startsWithEtcGmt) {
-            // Flip sign for Etc/GMT(+/-)H[H]
-            signChar = signChar == '-' ? '+' : '-';
-        }
+        return zoneId;
+    }
 
-        // extract the tens and ones characters for the hour
-        char hourTens;
-        char hourOnes;
-        if (length == 2) {
-            hourTens = '0';
-            hourOnes = zoneId.charAt(1);
-        }
-        else {
-            hourTens = zoneId.charAt(1);
-            hourOnes = zoneId.charAt(2);
-        }
+    private static String formatZoneOffset(int hour, int minute)
+    {
+        StringBuilder builder = new StringBuilder();
+
+        builder.append(hour >= 0 ? '+' : '-');
 
-        // do we have a valid hours offset time zone?
-        if (!isDigit(hourTens) || !isDigit(hourOnes)) {
-            return originalZoneId;
+        hour = abs(hour);
+        if (hour < 10) {
+            builder.append('0');
         }
+        builder.append(hour);
+        builder.append(':');
 
-        // is this offset 0 (e.g., UTC)?
-        if (hourTens == '0' && hourOnes == '0') {
-            return "utc";
+        if (minute < 10) {
+            builder.append('0');
         }
+        builder.append(minute);
 
-        return "" + signChar + hourTens + hourOnes + ":00";
+        return builder.toString();
     }
 
     private static boolean isUtcEquivalentName(String zoneId)
     {
-        return zoneId.equals("utc") ||
-                zoneId.equals("z") ||
-                zoneId.equals("ut") ||
-                zoneId.equals("uct") ||
-                zoneId.equals("gmt") ||
-                zoneId.equals("gmt0") ||
-                zoneId.equals("greenwich") ||
-                zoneId.equals("universal") ||
-                zoneId.equals("zulu");
+        return UTC_EQUIVALENTS.contains(zoneId.toLowerCase(ENGLISH));
```

Comment:
Does it matter we are calculating `zoneId.toLowerCase(ENGLISH)` twice now?

Reply:
Not really. I have another PR in progress to remove the case insensitivity. We don't support it, anyway.